plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,443 @@
+  }
+  THROW_IE_EXCEPTION << "Unexpected autopadding mode.";
+}
+
+// Extract values which is needed.
+edsl::Tensor extract_tensor(edsl::Tensor I, size_t rank) {
+  std::vector<TensorDim> I_dims(rank * (rank + 2) + 2), O_dims(rank + 2);
+  std::vector<TensorIndex> I_idxs(rank * (rank + 2) + 2), O_idxs(rank + 2);
+  I.bind_dims(I_dims);
+  // Set up the dimensions of output.
+  for (auto i = 2; i < rank + 4; ++i) {
+    O_dims[i - 2] = I_dims[i];
+  }
+  // Set up the indexs of batch_size and channel.
+  for (auto i = 0; i < 2; ++i) {
+    I_idxs[i + 2] = I_idxs[i];
+  }
+  // Set up the indexs of input.
+  for (auto i = 2; i < rank + 4; ++i) {
+    for (auto j = 1; j < rank; ++j) {
+      I_idxs[j * (rank + 2) + i] = I_idxs[i];
+    }
+  }
+  // Set up the indexs of output.
+  for (auto i = 2; i < rank + 4; ++i) {
+    O_idxs[i - 2] = I_idxs[i];
+  }
+  edsl::Tensor O = Contraction(O_dims, O_idxs).assign(I(I_idxs));
+  return O;
+}
+
+// Compute 1D DeformableConvolution.
+edsl::Tensor compute_1d_deformable_convolution(edsl::Tensor I, edsl::Tensor OFF, edsl::Tensor F,
+                                               std::vector<int64_t> I_shape, std::vector<int64_t> OFF_shape,
+                                               std::vector<int64_t> F_shape, int64_t G, int64_t DG, size_t rank,
+                                               std::vector<size_t> strides, std::vector<size_t> dilations,
+                                               std::vector<size_t> pad_befores, plaidml::op::AutoPadMode autopad_mode) {
+  // Define dim of each tensor.
+  auto N = I_shape[0], CI = I_shape[1], CO = F_shape[1], OFF_C = OFF_shape[1], W = I_shape[2], OFF_W = OFF_shape[2],
+       F_W = F_shape[2];
+  // Throw exception
+  if (CI % G != 0 || CO % G != 0 || CI % DG != 0 || OFF_C % DG != 0) {
+    THROW_IE_EXCEPTION << "Incorrected shape for DeformableConvolution.";
+  }
+  // Define new width.
+  auto NEW_W = OFF_W * F_W;
+  // split offset along dg axis and get offset of width.
+  std::vector<edsl::Tensor> offset_w_vec;
+  for (auto dg = 0; dg < DG; ++dg) {
+    edsl::Tensor OFF_split_dg =
+        op::slice(OFF).add_dim(0, N).add_dim(dg * OFF_C / DG, (dg + 1) * OFF_C / DG).add_dim(0, OFF_W);
+    edsl::Tensor OFF_split_dg_w = op::reshape(OFF_split_dg, make_tuple<int64_t>({N, 1, F_W, OFF_W}));
+    OFF_split_dg_w = op::transpose(OFF_split_dg_w, make_tuple<int64_t>({0, 1, 3, 2}));
+    OFF_split_dg_w = op::reshape(OFF_split_dg_w, make_tuple<int64_t>({N, 1, OFF_W * F_W}));
+    OFF_split_dg_w = op::broadcast(
+        OFF_split_dg_w, {static_cast<int>(N), static_cast<int>(CI / DG), static_cast<int>(NEW_W)}, {0, 1, 2});
+    offset_w_vec.push_back(OFF_split_dg_w);
+  }
+  edsl::Tensor offset_w = op::concatenate(offset_w_vec, 1);
+  // Define width index of input.
+  std::vector<edsl::TensorDim> F_dims(rank);
+  F.bind_dims(F_dims);
+  edsl::Tensor index_w = edsl::index({F_dims[2]}, static_cast<size_t>(0));
+  index_w = index_w * dilations[0];
```

Comment:
Can you use something like instead of the for loop, please check the output and change accordingly (2d index is like meshgrid in numpy):
```
 edsl::Tensor index_w = edsl::index({edsl::TensorDim(OFF_W), edsl::TensorDim(F_shape[2])]}, static_cast<size_t>(0));
 edsl::Tensor index_wT =  op::transpose(index_w, make_tuple<int64_t>({1,0}));
 index_w=index_wT*strides[0]+index_w*dilations[0]- pad_befores[0];
 index_w = op::reshape(index_w, make_tuple<int64_t>({1, 1, NEW_W}));
```